src/Cli/Microsoft.TemplateEngine.Cli/Commands/InteractiveMode/InteractiveMode.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Edge.Settings;
+
+namespace Microsoft.TemplateEngine.Cli.Commands.InteractiveMode
+{
+    public static class InteractiveMode
+    {
+        public static async Task EnterInteractiveMode(InvocationContext context, Func<InvocationContext, Task> next)
+        {
+            // TODO: make sure we have access to Instantiate command
+            if (context.ParseResult.CommandResult.Command.Name == "new" || context.ParseResult.CommandResult.Command.Name == "create")
+            {
+                InstantiateCommand? instantiateCommand = GetInstantiateCommand(context.ParseResult.CommandResult.Command);
+                if (instantiateCommand is null)
+                {
+                    return;
+                }
+
+                var enterInteractiveMode = ShouldEnterInteractiveMode(context.ParseResult);
+                if (enterInteractiveMode)
+                {
+                    // TODO: add the default skip key
+                    // TODO: short name validation. right now we are assuming that the provided template exists and is valid
+                    context.Console.WriteLine("Currently under development, not all features implemented yet");
+
+                    // --- Workaround to get the environmental settings at this stage. Firgure this out later
+                    ITemplateEngineHost host = instantiateCommand.Host(context.ParseResult);
+                    IEnvironment environment = new CliEnvironment();
+                    IEngineEnvironmentSettings settings = new EngineEnvironmentSettings(
+                        host,
+                        environment: environment,
+                        pathInfo: new CliPathInfo(host, environment, null));
+
+                    TemplatePackageManager templatePackageManager = new(settings);
+                    HostSpecificDataLoader? hostSpecificDataLoader = new(settings);
+
+                    IReadOnlyList<ITemplateInfo> templates =
+                        Task.Run(async () => await templatePackageManager.GetTemplatesAsync(default).ConfigureAwait(false)).GetAwaiter().GetResult();
```

Comment:
why not just `await`? This method is async

Reply:
Method was not async at the beginning, this is just a bit of overlooked code when I made it async!